cobc/cobc.c

```diff
@@ -4434,7 +4458,8 @@ process_filename (const char *filename)
 		fn->translate = cobc_main_strdup (output_name);
 	} else
 	if (save_all_src || save_temps || save_c_src
-	 || cb_compile_level == CB_LEVEL_TRANSLATE) {
+	    || cb_include_file_list
```

Comment:
Is there a reason that we need the change here? Please explain or revert that line.

Reply:
If we don't do that, `#include "file.h"` will be compiled in `/tmp/`, so it will lookup `file.h` in `/tmp/` instead of in the project.

Reply:
Wouldn't it be more reasonable to auto-add `-I .` to the used compiler flags or doesn't this work?

Reply:
What's the problem with the current behavior ? Is it because it might pollute the current directory with binary artefacts ?

Asking the user to use `-I $PWD` would also have an impact on `#include <...>` macros, when the option is only adding a quote include, that is supposed to be the only ones looking for files inside the project.

Reply:
> Is it because it might pollute the current directory with binary artefacts ?

This is one of the reasons. Another one is that those artifacts may even overwrite actual source files (would be troublesome in any case, but we do have an issue about this in our trackers for a reason).

On the second thought I agree that **auto-adding** `-I .` was a bad idea, thank you for not jumping on that train.
It is up to the user to decide how he handles that, the one that will likely be _most_ used is on Win32 with `--include="<windows.h>"` or `--include="<pdcurses.h>"`  (= files that are in the system default include paths already, also on other systems).
Let's drop the _auto-adding_ of `--save-temps` and in the testsuite use `--include="$PWD/file.h"`.

Reply:
> Let's drop the _auto-adding_ of `--save-temps` and in the testsuite use `--include="$PWD/file.h"`.

But there is no auto-adding of `--save-temps`, this test only modifies the location of the generated C name, i.e. the current directory instead of the `/tmp` directory. That's what you mean ?

Reply:
I see, it's only a partial version and then deletes the file after use.

But you're fine with not adding any path and just let the user specify `--include="$PWD/file.h"`, if this is what he wants, right? In this case we won't need that the C file to be generated locally, either (as noted, most often people actually want to use a system-installed header).

Reply:
What troubles me with the needed $PWD is that it’s not the same behavior as in gcc, most people will get an error the first time unless they check the doc first.

Reply:
I've thought about that... and take full responsibility. Please check in with $PWD, then this is ready for svn.